plugins/imessage.sma

```diff
@@ -14,110 +14,133 @@
 #include <amxmodx>
 #include <amxmisc>
 
-#define X_POS         -1.0
-#define Y_POS         0.20
-#define HOLD_TIME     12.0
+const MAX_MSG_LEN = 384;
+const TASK_MSG    = 12345;
 
-new Array:g_Values
-new Array:g_Messages
-new g_MessagesNum
-new g_Current
+enum _:MessageInfo
+{
+	Message[MAX_MSG_LEN],
+	R,
+	G,
+	B
+}
+
+new Array:g_messages;
+new g_messagesNum;
+new g_hudObject;
+new g_current;
 
-new amx_freq_imessage;
+new g_hostname[64];
+new Float:g_amx_freq_imessage;
+new Float:g_amx_imessage_x_pos;
+new Float:g_amx_imessage_y_pos;
+new Float:g_amx_imessage_holdtime;
+new g_amx_imessage_only_dead;
 
 public plugin_init()
 {
-	g_Messages=ArrayCreate(384);
-	g_Values=ArrayCreate(3);
-	register_plugin("Info. Messages", AMXX_VERSION_STR, "AMXX Dev Team")
-	register_dictionary("imessage.txt")
-	register_dictionary("common.txt")
-	register_srvcmd("amx_imessage", "setMessage")
-	amx_freq_imessage=register_cvar("amx_freq_imessage", "10")
-	
-	new lastinfo[8]
-	get_localinfo("lastinfomsg", lastinfo, charsmax(lastinfo))
-	g_Current = str_to_num(lastinfo)
-	set_localinfo("lastinfomsg", "")
+	register_plugin("Info. Messages", AMXX_VERSION_STR, "AMXX Dev Team");
+	register_srvcmd("amx_imessage", "setMessage", _, "<message> <color in RRRGGGBBB format>");
+
+	register_dictionary("imessage.txt");
+	register_dictionary("common.txt");
+
+	g_messages = ArrayCreate(MessageInfo);
+	g_hudObject = CreateHudSyncObj();
+
+	bind_pcvar_num(create_cvar("amx_imessage_only_dead", "0", _, "Set to 1 to show info messages only to dead clients", true, 0.0, true, 1.0), g_amx_imessage_only_dead);
+	bind_pcvar_float(create_cvar("amx_freq_imessage", "180", _, "Frequency in seconds of info messages", true, 0.0), g_amx_freq_imessage);
+	bind_pcvar_float(create_cvar("amx_imessage_x_pos", "-1.0", _, "X position for info messages", true, -1.0, true, 1.0), g_amx_imessage_x_pos);
```

Comment:
Bound logic doesn't fit good here, because the valid values are `0.0 <= x <= 1.0 || x == -1.0`.